src/Widgets/DisplaysOverlay.vala

```diff
@@ -186,10 +186,12 @@ public class Display.DisplaysOverlay : Gtk.Box {
 
     public void rescan_displays () {
         scanning = true;
-        foreach (unowned var widget in display_widgets) {
-            display_widgets.remove (widget);
-            widget.destroy ();
-        }
+
+        display_widgets.@foreach ((display_widget) => {
+                overlay.remove_overlay (display_widget);
+                display_widget.destroy ();
+                display_widgets.remove (display_widget);
+        });
```

Comment:
We seem to have extra indentation here.

```suggestion
        display_widgets.@foreach ((display_widget) => {
            overlay.remove_overlay (display_widget);
            display_widget.destroy ();
            display_widgets.remove (display_widget);
        });
```

BTW I think `foreach (unowned var widget in display_widgets)` instead of `.@foreach` should still work here though. We don't use `@foreach` in this file elsewhere.

Reply:
`foreach (unowned var widget in display_widgets)` interects only one time if you have two or more displays here, I know it happens because of `remove ()`, but I don't know exactly why, I think the cause can be if you remove data from the first node, then it's get freed, then `foreach` don't have the next pointer reference and ends execution.

Reply:
We could access to the list in the C-style as you write, but maybe we should keep `@foreach ()` which is more Vala-ish if it works as expected then.